lib/topologies/shared.js

```diff
@@ -402,6 +405,36 @@ const SessionMixins = {
   }
 };
 
+const RETRYABLE_WIRE_VERSION = 6;
+
+/**
+ * Determines whether the provided topology supports retryable writes
+ *
+ * @param {Mongos|Replset} topology
+ */
+const isRetryableWritesSupported = function(topology) {
+  const maxWireVersion = topology.lastIsMaster().maxWireVersion;
+  if (maxWireVersion < RETRYABLE_WIRE_VERSION) {
+    return false;
+  }
+
+  if (!topology.logicalSessionTimeoutMinutes) {
+    return false;
+  }
+
+  return true;
+};
+
+/**
+ * Increment the transaction number on the ServerSession contained by the provided ClientSession
+ *
+ * @param {ClientSession} session
+ */
+const txnNumber = function(session) {
```

Comment:
Do we need to abbreviate this to `txnNumber`? Can we name it `transactionNumber`, or better `getNextTransactionNumber`?